packages/ckeditor5-editor-multi-root/src/multirooteditoruiview.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module editor-multi-root/multirooteditoruiview
+ */
+
+import { EditorUIView, InlineEditableUIView, ToolbarView } from 'ckeditor5/src/ui';
+import type { Locale } from 'ckeditor5/src/utils';
+import type { View } from 'ckeditor5/src/engine';
+
+/**
+ * The multi-root editor UI view. It is a virtual view providing an inline
+ * {@link module:editor-multi-root/multirooteditoruiview~MultiRootEditorUIView#editable} and a
+ * {@link module:editor-multi-root/multirooteditoruiview~MultiRootEditorUIView#toolbar}, but without any
+ * specific arrangement of the components in the DOM.
+ *
+ * See {@link module:editor-multi-root/multirooteditor~MultiRootEditor.create `MultiRootEditor.create()`}
+ * to learn more about this view.
+ */
+export default class MultiRootEditorUIView extends EditorUIView {
+	/**
+	 * The main toolbar of the multi-root editor UI.
+	 */
+	public readonly toolbar: ToolbarView;
+
+	/**
+	 * Editable elements used by the multi-root editor UI.
+	 */
+	public readonly editables: Record<string, InlineEditableUIView>;
+
+	public readonly editable: InlineEditableUIView;
+
+	/**
+	 * Creates an instance of the multi-root editor UI view.
+	 *
+	 * @param locale The {@link module:core/editor/editor~Editor#locale} instance.
+	 * @param editingView The editing view instance this view is related to.
+	 * @param editableNames Names for all editable views. For each name, one
+	 * {@link module:ui/editableui/inline/inlineeditableuiview~InlineEditableUIView `InlineEditableUIView`} instance will be initialized.
+	 * @param options Configuration options for the view instance.
+	 * @param options.editableElements The editable elements to be used, assigned to their names. If not specified, they will be
+	 * automatically created by {@link module:ui/editableui/inline/inlineeditableuiview~InlineEditableUIView `InlineEditableUIView`}
+	 * instances.
+	 * @param options.shouldToolbarGroupWhenFull When set to `true` enables automatic items grouping
+	 * in the main {@link module:editor-multi-root/multirooteditoruiview~MultiRootEditorUIView#toolbar toolbar}.
+	 * See {@link module:ui/toolbar/toolbarview~ToolbarOptions#shouldGroupWhenFull} to learn more.
+	 */
+	constructor(
+		locale: Locale,
+		editingView: View,
+		editableNames: Array<string>,
+		options: {
+			editableElements?: Record<string, HTMLElement>;
```

Comment:
There are 2 params (`editableNames` and `options.editableElements`) that could be merged into one (non optional) parameter (with optional values in the record). To make it consistent with [`InlineEditorUIView`](https://github.com/ckeditor/ckeditor5/blob/c8a89f383efcd7d6369012039b2db6409d419467/packages/ckeditor5-editor-inline/src/inlineeditoruiview.ts#L137-L144) I would suggest using top level param (not in the options).

Reply:
This was my initial approach. After a while I came up with this which was easier to use and to describe in docs. It also makes sense to me, when compared to other `EditorUIView`s where `editableElement` is optional and passed in options.

Reply:
:+1: